scripts/table_updates/update_data_element_catalog.py

```diff
@@ -71,125 +101,221 @@ def _get_syn_col_type(var_type, validation):
     else:
         return "STRING"
 
-def _create_new_row(df, cohort):
+
+def _create_new_row(df: pandas.DataFrame, cohort: str) -> pandas.DataFrame:
     """
-    Create new row for updating the data element catalog
+    Add new rows for variables introduced in the specified version of the data dictionary to the data element catalog table.
+
+    Args:
+        df:  The dataframe contains the variables to be added.
+        cohort: Name of the cohort associated with the data dictionary.
+
+    Returns:
+        pandas.DataFrame: A new row containing the necessary columns for the data element catalog.
     """
-    # add new and update old variables to data element catalog on Synapse
     # add: variable, instrument, dataType='curated', type, label, cohort_dd, synColType, synColSize, numCols, colLabels
-    df['synColType'] = df.apply(lambda row: _get_syn_col_type(row.type,row.validation), axis=1)
-    df_choices = df[df['type'].isin(['dropdown','radio','checkbox'])]
-    df_choices.loc[:, ['choices_num','max_len','choices_key']] = df_choices['choices'].apply(_get_choices_info)
-    # df_choices[['choices_num','max_len','choices_key']] = df_choices['choices'].apply(_get_choices_info)
-    non_checkbox_index = df_choices.index[df_choices.type.isin(['dropdown','radio'])]
+    df["synColType"] = df.apply(
+        lambda row: _get_syn_col_type(row.type, row.validation), axis=1
+    )
+    # Calculate the new choices_num, max_len, and choices_key for variables with choices
+    df_choices = df[df["type"].isin(["dropdown", "radio", "checkbox"])]
+    df_choices.loc[:, ["choices_num", "max_len", "choices_key"]] = df_choices[
+        "choices"
+    ].apply(_get_choices_info)
+    non_checkbox_index = df_choices.index[df_choices.type.isin(["dropdown", "radio"])]
     checkbox_index = df_choices.index[df_choices.type == "checkbox"]
-    df.loc[non_checkbox_index,'synColSize'] = df_choices.loc[non_checkbox_index,'max_len']
+    # for non-checkbox type, update synColSize only since no new columns will be added
+    if len(non_checkbox_index) > 0:
```

Comment:
Add conditions to update non_checkbox column size to align with checkbox columns.